### ISSUE TITLE
Lack of Clarity on Authorized Activities

### ISSUE THREAD

**Issue by zoffixznet** [perl6/user-experience#33]
It's not at all clear to me what constitutes an authorized action, in the context of performing core work for the language.

For the past several months, I have worked on preparing the 6.d specification release, utilizing the repositories access I've been given by the TPF and publicly-visible (and dev-chat bot-announced) TODO lists for the release in [6.d-prep](https://github.com/perl6/6.d-prep) repository.

Upon completion of that work, a leading Rakudo core developer and member of the TPF Grants Committee [has started](https://liztormato.wordpress.com/2018/11/06/on-raku/) a [campaign](https://meta.stackoverflow.com/a/376270) to tarnish my reputation and devalue the work I've done in those months.

She is under the impression that I performed the work with malicious intent and without proper authorization. Thus, I would like the Management to clarify the proper channels the core developers have to follow to obtain appropriate authorization for the work they plan on doing.

**Comment by lizmat** [comment]
@zoffixznet: at the Amsterdam.PM meeting now until late, then have a 2+ hour drive ahead of me.  So please don't expect any response from me for at least 14 hours.

**Comment by lizmat** [comment]
Re: [I tried to bring up my concerns to the people in charge, but lizmat appears to claim that she's in charge, so I don't expect any clarifications to come through.](https://www.reddit.com/r/perl6/comments/9uqsxl/a_warning_to_the_community/e986r73/)

Oh, how I wish I were in charge of Perl 6.  I would have been able to say "no" against the current way the "alias" thing is working out.  That would have saved a lot of effort.

Meanwhile, I think we're all suffering from lack of clarity.  A clarity, which I'm afraid, will not come from Larry Wall.

@zoffixznet: although we disagree on the way the "alias" should be implemented, we both think that something should happen now that this project is growing up.  If the project would benefit from my absence, and this is asked of me, then I will leave and wish you all the best.  Sometimes you have to leave the thing you love alone so it can thrive on its own.

@stmuk: please don't remove this comment on grounds that it doesn't belong on Github.

**Comment by CIAvash** [comment]
I ask all sides to remain calm.

I think the main problem is somewhere else, I'm not sure what it is, but there is something wrong with the community. Maybe lack of communication, miscommunication, emotionally reacting to everything, taking things personally, always being on the offence, personal attacks, ...

**Comment by stmuk** [comment]
I'm looking at https://design.perl6.org/S01.html 

"Mostly, we're just a bunch of ants all cooperating (sort of) to haul food toward the nest (on average). There are many groups of people working on various bits and pieces as they see fit, since this is primarily a volunteer effort."

@lizmat you aren't perl leader you are an ant like everyone else.

Even if you were BDFL it's clear its not about forcing people to do things.

"The language designer is neither omniscient nor omnipotent, and never will be, despite requests for those particular features. Therefore the design process will be spiral, cooperative, and convergent. The rate of convergence is an emergent property, and cannot be forced, only encouraged... If you are unhappy with the current rate of convergence, please cooperate more with someone else you think is interested in convergence. "

**Comment by lizmat** [comment]
> its not about forcing people to do things

Indeed, it isn't.  It's about vision and providing a framework in which people can thrive to the betterment of the project.  The project has a very weak framework.  When people shake the framework, cracks become obvious.  Looks like at least two people are now exposed to the elements.

Again, until someone with a commit bit asks me to leave, I will stay around to help the project.

**Comment by cjfields** [comment]
As a part-time user and now full time group leader, I find the best solutions tend to rise from conflict, to be honest.  But IMHO I do think there needs to be a bit of a cooling off period.

**Comment by stmuk** [comment]
If the project had a weak framework it would have collapsed before. We continue to muddle through.

Noone should leave. We need all the leaf haulers we can get!

**Comment by zoffixznet** [comment]
> lack of clarity

Yes.

> something wrong with the community

Given the reaction this week to what I viewed as my routine work in the organization, it's clear to me there are a lot of people who have a lot of problems with how some people do things. And the problem is instead of voicing those concerns with clear argumentation, people hide those feelings because we're a "nice" community and instead they share those concerns only in their little cliques.

I don't have to look far for examples, a mere 1.5hr ago [a user indicated a DM was sent](https://colabti.org/irclogger/irclogger_log/perl6?date=2018-11-07#l485). I'm 90% sure the DM is about holyghost's modules. A negative message about that user has been passed around to a couple of people so far, including myself. Yet, no one talks about the problems with the modules directly with holyghost.

Yesterday, I found out flussence wrote a whole tirade about me and how I've been "wrecking" everything since 2015. That's three years, yet the only communication I remember from that user was about prioritizing implementation of `pack`.

I can only surmise why ugexe is pissed with me. I've no idea why tony-o is hostile towards me.

No one talks about issues directly with the people causing them. It's always bottled up until it explodes all over, with people writing accusatory blogs and calling each other names.

Even you, @lizmat,  [when you had the opportunity](https://colabti.org/irclogger/irclogger_log/perl6-dev?date=2018-11-04#l518) to discuss an issue that bothers you, you instead exited the channel. If you would've said at that point that the current direction strongly bothers you and you'd like TimToady to clarify the appropriate use of the alias, we would have simply changed the brochure to say "An alias has been chosen. We're waiting for Larry Wall to finalize how it's meant to be used." That's it. A simple solution. Instead, you and your wife started spreading paranoia about how I'm coup d'etating and a full rename has occurred.

> If the project would benefit from my absence, and this is asked of me, then I will leave and wish you all the best.

The promise of this project when I joined was [`-Ofun`](https://perl6.org/fun/). When thinking today, I realized that for the past few months, I've neglected a lot of personal life for 6.d spec release work: avoiding doing homework for my French classes, skipping my guitar practice, skipping gym, postponing an *Elder Scrolls Online* quest—it was not `-Ofun`, but I knew the spec release had to be done and I knew it would be `-Ofun` to finally have the 6.d release is done.

Turns out it wasn't. It's not `-Ofun` to see people trash your work behind your back. It's not `-Ofun` to have people call you names. It's not `-Ofun` to have people discredit your work. It's not `-Ofun` to see people spread lies. It's not `-Ofun` to be working on a commit, while wondering whether someone is sending someone else a DM about it, instead of talking about the problem directly.

I think I need to take a long look at my life and re-evaluate what is in it that actually makes me happy.

As of this moment, I'm taking a hiatus to find that answer.

**Comment by stmuk** [comment]
@zoffixznet You did a superb job with 6.d and it sounds like you deserve a break. I hope you return in the near future!

**Comment by lizmat** [comment]
> @zoffixznet You did a superb job with 6.d and it sounds like you deserve a break.

Accepting the possibility of being called a hypocrite, I would like to express that I fully agree with that sentiment.  We disagree on the naming alias.

>  I hope you return in the near future!

And that it once again be -Ofun for you rather than unpaid $work.

**Comment by tony-o** [comment]
TL;DR either rename the language or continue calling it perl6

@zoffixznet if you believe me to be hostile then it's your own feelings taking part in reading my messages, I don't write to you or many people in IRC or here with emotion and I certainly don't waste any emotion on this topic.

What I do see is you talking out of both sides of your mouth on the issue and dismissing @lizmat's legitimate concerns with `it's just an alias` instead of addressing her concerns directly as though your intention is *not* to rename the entire language ([see here](https://colabti.org/irclogger/irclogger_log/perl6?date=2018-10-08#l757)) after "proving" an alias will be superior and then immediately setting to work to change materials to reflect the alias after concerned voices are raised asking for clarification with BFDL sees the alias being used.  Shit or get off the pot, either rename the language or keep calling it `perl6` and stop talking about it - everyone loses when optimizations don't make it into a release because we're arguing a name.

For the people who keep saying "it's just a (compromise|alias)" this is where the "it's a thinly veiled rename" theme arose and continues to prevail, it's in the intent of creating that compromise in the first place.

The other part that is becoming more apparent is that you, directly, have caused a lot of turmoil in the community and then keep claiming you're going on hiatus.  If you're going to make changes that you know will polarize a community, then stay and face the weather or don't make the change.  @niner has direct experience with that (and I was one of the louder critics) and we all benefited from him staying around.  I don't know how he considers our interactions but I think at PTS they've been fruitful and fun.

The last point I'll make is that `perl6` won common vote [here](https://www.perlmonks.org/?node_id=1217692) and the option to not create an alias, or drop the idea, was not even considered.  That doesn't seem like a compromise to me but an intentional strong arm from someone in a position to make those changes.  I'll continue to use the phrase `badgering timtoady` to describe how this alias came to be.

Enjoy your time away and hopefully things become fun for you again.

**Comment by stmuk** [comment]
@tony-o The alias discussion is over. Larry ruled. You lost.

**Comment by tony-o** [comment]
@stmuk not being able to read must be a skill.  I don't care about the alias, I care about what it is doing the community.

**Comment by stmuk** [comment]
@lizmat It's possible he may change his mind but unlikely given he said "Final Answer". He's a linguist who likes words and likes "Raku" (as I do).  We shall see.  

Your favoured version differs only by the addition of two letters. What an awful lot of fuss and bother there has been over that  "D" and "O"! Is it really worth it I wonder?

**Comment by bazzaar** [comment]
Some very condescending language used in that farewell note, it seems to me, and some of the rest maybe designed to spread yet more division. An undeniable amount of good work, and yet the source of community strife, and the subsequent damage to the image of Perl 6. The intent seemed to be to loosen the grip that Perl has on 'Perl 6' [admittedly from my perpective on the periphery]. Why?

**Comment by tony-o** [comment]
@stmuk a fan of words might also interpret `final answer` to be in response to the back and forth about _which_ alias he'd prefer and not for the discussion itself.  In addition to that, point 2 makes further discussion worthwhile for the side that "lost"

**Comment by lizmat** [comment]
@stmuk: let me [repeat my stance on renaming](https://www.reddit.com/r/perl6/comments/9m2k6l/a_request_to_larry_wall_to_create_a_language_name/e7cdgih/)

I emphasize: **The name is correct as it is, one can refer to "Rakudo" to indicate a specific implementation of Perl 6, just as one can refer to "pugs" or "Fanlang" or "Niecza".**

Also: **There is no other marketing name needed.**

Saying there is a new "Rakudo Perl 6 Compiler Release", means that the **Rakudo implementation** of Perl 6 has a new release.  This does **not** entail changing any documentation / marketing / tagging of questions.

> Your favoured version differs only by the addition of two letters. What an awful lot of fuss and bother there has been over that "D" and "O"! Is it really worth it I wonder?

To you and me it apparently is worth having words about.  What can I say?  I felt strongly enough about it to write a blog post, indicating that this was **MY** opinion.  I've also made clear that if anybody with a commit bit thinks I should leave the project, I will.

Sometimes things matter.  Especially when they only appear to be as little as two letters difference.  When in fact there is a much, much more important thing hiding behind those letters.

**Comment by Leont** [comment]
Quite frankly nothing good is coming out of this discussion. Emotions are clearly too high for any kind of meaningful conversation. Github is not the right place to blow off steam.

I'm closing this ticket. I will reopen it in exactly 7 days (14 nov 20:00 GMT). 

In the mean time, please hug your loved ones, go out for a walk with your dog, and have a drink with some fellow perl mongers.

**Comment by shadowcat-mst** [comment]
Seconding. See also: https://twitter.com/shadowcat_mst/status/1059859981921607682

**Comment by jnthn** [comment]
Yes, I know this was just closed, but a) the original question wasn't addressed, b) I've been planning to answer since it was asked, and c) I'm the pumpking, and the original question was, by asking "the management", effectively addressed to one of myself or Larry.

In this reply I want to address the following points (and *nothing* more):

> It's not at all clear to me what constitutes an authorized action, in the context of performing core work for the language.

...

> Thus, I would like the Management to clarify the proper channels the core developers have to follow to obtain appropriate authorization for the work they plan on doing.

The community around the Perl 6 language has primarily operated thus far around the principles of "forgiveness over permission" and that, given the opportunity to contribute, the overwhelming majority of people will do so in positive and productive ways. This is demonstrated, for example, in a generally very liberal commit bit policy.

There's also been an intentional effort to *not* have most things that happen require an OK from an "authority", "management", or however one might wish to term it. Clearly, there's no way two otherwise busy people would have the time to micro-manage an effort as vast as what the Perl 6 community is doing. But - and I won't put words in Larry's mouth here, and only speak for myself - my feeling is that even *were* time available to do that, it still wouldn't be a wise choice; we'd miss out on all the things, small and large, that folks just do because they feel they might be valuable, but might not consider if they had to "ask management" first.

Of course, such an approach makes some demands on all who participate. There's an expectation that work that will have a sizable impact on Perl 6 language users and contributors will be carried out transparently, and that effort will be made to build consensus around it (or to make sure that the consensus is already there).

It goes without saying that the 6.d release of the language is a major undertaking, and is making commitments that those producing Perl 6 implementations will be expected to keep. So far as I'm concerned, the 6.d process was, overall, in keeping with what I'd expect for something of this magnitude. A rough target for timescale and scope was put forward far in advance, and progress on the enormous task of reviewing all of the commits to the specification test suite was reported regularly on the `#perl6-dev` channel. There was broad consensus that a 6.d release was timely - perhaps even overdue - and there's no arguing that "the management" didn't approve, since I - multiple times - expressed my appreciation to Zoffix for the ongoing work he was doing towards 6.d.

While consensus is preferable, there are issues where that fails, and somebody has to make a decision (this isn't perfect, but "design by committee" would seem to be even less perfect). For Rakudo/MoarVM issues, that presently falls on me. For language design issues, that ultimately falls on Larry, although nowadays I'm also providing some rulings on smaller language design matters, with the proviso that Larry can show up and say I'm wrong. It goes without saying that a language name alias is not a "small matter".

Whatever one may think of the alias decision, the fact of the matter is that the request for one was presented in an organized manner after an amount of research and with usage intent clearly stated. (I wish more language design proposals were submitted with that level of preparation!) Of note, it was clear in that proposal that the alias was being considered as a 6.d deliverable *and* it was explicitly stated in the request that there was intent to use it in informational material:

> I ask that Larry Wall renders his decision on the alias by November 1, 2018, so > we would have the time to create proper informational materials for the 6.d
> language release, during which time, the name alias would be officially
> announced, if one is chosen.

It was therefore reasonable to assume that, with Larry's choice of an alias, the activity of mentioning it in the 6.d release information material was an OK thing to do.

In summary, the general principle is: try to do it by consensus, and if that fails, ask whichever of myself or Larry you consider most appropriate for the question at hand. So far as the specific example of 6.d release activities goes, there was sufficient consensus around the 6.d release for all elements except the alias, and the use of the alias in informational materials was a reasonable action based on the rendered decision, especially owing to its explicit mention in the initial request for a decision.

And, a reminder of what I wrote above: this is *solely* addressing the original question about authorization and process. It should not be taken as a position on any other matter. With the original question answered, this issue can be considered closed for good.

**Comment by AlexDaniel** [comment]
This is now resolved. The [problem-solving](https://github.com/perl6/problem-solving/) repo improves the situation in at least two ways:
1. Impactful changes require approval from all listed devs, meaning that everyone will now be aware of the upcoming changes and they can ask for clarifications if needed
2. There are assigned people who work on certain areas

We'll see how well it works in practice, but I'm convinced it'll work significantly better than nothing.

I miss you, @zoffixznet.

**Comment by AlexDaniel** [comment]
Ah, one downside that I can see right now is that in some cases it feels like doing extra paperwork, and it's a bit hard to get people to follow the process. I'll think how that can be improved, but it's better than having problems outlined in this ticket.

**Comment by AlexDaniel** [comment]
Also, if anybody thinks that there are still issues mentioned in this ticket that are not resolved, please either talk to me or create a ticket [here](https://github.com/perl6/problem-solving/issues).
